client/app/components/WithIcon.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+
+export interface WithIconProps {
+  children?: string;
+  icon: JSX.Element;
+  alt: string;
+}
+
+export function withIcon<ComponentProps extends object>(Component: React.ComponentType<ComponentProps>) {
```

Comment:
The proposal of this component is to become the standard approach to icons, as it is accessible by default. Soon this will be documented.

Reply:
What's the advantage of having

```
const Component = withIcon(MyComponent);
return <Component {...props} />
```
versus
```
return <IconText wrapper={Component} icon={MyIcon} {...props}>;
```

where `wrapper` is `span` by default?

Ant's got some icon text stuff (not exactly this) -- might be nice if our approach roughly matched theirs, as long as we're using ant.

https://ant.design/components/typography/